theano/misc/pkl_utils.py

```diff
@@ -275,25 +275,31 @@ class PersistentNdarrayLoad(object):
     """
     def __init__(self, zip_file):
         self.zip_file = zip_file
+        self.cache = {}
 
     def __call__(self, persid):
         array_type, name = persid.split('.')
 
+        if name in self.cache:
+            return self.cache[name]
```

Comment:
Because the codes are using **name** instead of **type.name** as filename to store data in **zip_file**. You can see them at Line 195 and Line 208. A **persid** with same **name** must comes from same file named by **name**. So I use **name** instead of persid in case there are two different persid with same name, though maybe it is impossible.